src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - The `die` method uses `indexOf` and `splice` to remove the animal from `Animal.alive`. This can be inefficient if there are many animals, as it requires searching through the array and then reindexing it after removal. Consider using the `filter` method to remove the dead animal more efficiently.